### PR TITLE
Widgets: Refactor the inspector in Widgets Customizer to use core's controls

### LINKED ISSUE #30271: The inspector in the widgets customizer should follow accessibility best practices
- Pressing <kbd>esc</kbd> should close the inspector.
- Scrolling in the inspector should trigger the header to be sticky.
- Opening the inspector should move focus to its first focusable element (the back button)

_Originally posted by @kevin940726 in https://github.com/WordPress/gutenberg/issues/29755#issuecomment-801610914_

### PR DESCRIPTION
**Pull request by kevin940726**
## Description

Supersede #29755, close #30271.

I essentially rewrote the inspector by using core's controls section. Simplified the code a bit and moved most of the logics to `controls`. This also fixes the accessibility and UX issues of the previous attempt.
 
## How has this been tested?

1. Enable "Widgets blocks in Customizer" gutenberg experiment flag.
2. Go to Customize -> Widgets.
3. Inspect some blocks, it should open the inspector panel in a new section.
4. The focus should be moved to the back button of the inspector section.
5. Clicking on it or pressing <kbd>Escape</kbd> should close the inspector.
6. Closing the inspector should smoothly transition back to the editor and return focus back to the selected block.
7. It's possible to go to different widget areas and open their corresponding inspector panel.
8. Scrolling in the inspector should trigger the sticky header (See the screen recordings for more details.)
9. The inspector now has the correct title.

## Screenshots 

https://user-images.githubusercontent.com/7753001/113235395-890dbb80-92d5-11eb-91ae-106c8df0d164.mp4

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/customize-widgets/src/components/sidebar-block-editor/index.js [hunk 1]
```diff
@@ -69,52 +44,46 @@ export default function SidebarBlockEditor( { sidebar, inserter } ) {
...
 					{ createPortal(
-						<Inspector
-							isOpened={ isInspectorOpened }
-							isAnimating={ isInspectorAnimating }
-							setInspectorOpenState={ setInspectorOpenState }
-						/>,
-						parentContainer
+						// This is a temporary hack to prevent button component inside <BlockInspector>
+						// from submitting form when type="button" is not specified.
+						<form onSubmit={ ( event ) => event.preventDefault() }>
```
Comment (kevin940726) [inline packages/customize-widgets/src/components/sidebar-block-editor/index.js:76]: I opened https://github.com/ItsJonQ/g2/pull/298 to track this.

**Comment by noisysocks** [comment]
I like this approach better 🙂

**Comment by noisysocks** [comment]
I noticed two bugs:

- The slide out inserter's header looks all garbled.

  Screen Shot 2021-04-07 at 10 23 06

- If I open an inspector, go back two levels, then click on Widgets, the transition is sluggish / weird. I tried to screen record this but the screen recording software made the transition impossible to see.

**Comment by kevin940726** [comment]
> The slide out inserter's header looks all garbled.

This is an oversight, fixed in c81cbab.

> the transition is sluggish / weird.

This is a bit weirder, it seems like it has something to do with React remounting the tree twice. Since currently I can't get 2+ block editors work nicely together at the same time on the page, I unmount the previous one whenever it's collapsed and remount it when expanded. I delay the mounting to after the transitioning to avoid the weird transitioning effect, also make it more responsive to user's interactions. If it's not solvable (rendering multiple block editors) at the time, then maybe we should consider adding some loading screen when transitioning 🤔.

**Comment by noisysocks** [comment]
> If it's not solvable (rendering multiple block editors) at the time

It should be possible. The idea of a `@wordpress/block-editor` is that it's embeddable. For example, a P2 blog has one block editor per _Write comment_ form. Do you know what the problem is when two block editors are mounted on the same page? Perhaps we can isolate the problem by creating a minimal HTML page which has two block editors?

**Comment by kevin940726** [comment]
> a P2 blog has one block editor per _Write comment_ form.

Oh I didn't know that!

> Do you know what the problem is when two block editors are mounted on the same page? Perhaps we can isolate the problem by creating a minimal HTML page which has two block editors?

Yes we should be able to, but I think that's up to a different PR. The performance concern is crucial too. I would imagine mounting multiple fully-featured block editors has some serious performance impact, but I need to do some tests to make sure, maybe it's not that bad.

File: packages/customize-widgets/src/components/sidebar-block-editor/index.js (hunk 1 above)
Comment (noisysocks) [reply packages/customize-widgets/src/components/sidebar-block-editor/index.js:76]: That PR is merged now. Can we remove the hack?

**Comment by noisysocks** [comment]
> If it's not solvable (rendering multiple block editors) at the time, then maybe we should consider adding some loading screen when transitioning 🤔.

Yeah, I see what you mean here. I am definitely not a fan of how the editor "pops in" now _after_ the transition is complete. It makes the UI feel very unresponsive. If it is not feasible to mount the entire editor in the 16 ms that we have before the transition starts then we should try to mount only some of the block editor. Perhaps only the block editor's chrome but not the blocks.

But, yes, this can be done in a follow-up 🙂

**Review by noisysocks — APPROVED** [review]

File: packages/customize-widgets/src/components/sidebar-block-editor/index.js (hunk 1 above)
Comment (kevin940726) [reply packages/customize-widgets/src/components/sidebar-block-editor/index.js:76]: It's not yet being merged into trunk yet.

**Pull request merged**